lib/phoenix_live_view/channel.ex

```diff
@@ -167,15 +181,15 @@ defmodule Phoenix.LiveView.Channel do
     end
   end
 
-  defp maybe_call_mount_handle_params(%{socket: socket} = state, router, url, params) do
+  defp maybe_call_mount_handle_params(%{socket: socket} = state, url, params) do
     %{view: view} = socket
 
     cond do
       not function_exported?(view, :handle_params, 3) ->
         {diff, new_state} = render_diff(state, socket)
         {:ok, diff, :mount, new_state}
 
-      is_nil(router) ->
+      socket.root_pid != self() ->
```

Comment:
I think we need to do this:

```suggestion
      socket.root_pid != self() or is_nil(router) ->
```

After all, live patch only works if the root live view has a router.